packages/jsii/lib/compiler.ts

```diff
@@ -72,28 +72,70 @@ export class Compiler implements Emitter {
   }
 
   /**
-     * Compiles the configured program.
-     *
-     * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
-     */
+   * Adds a listener for the 'result' event that is called with EmitResult.
+   *
+   * @param event    the 'result' event
+   * @param listener the listener to be called with the EmitResult.
+   */
+  public on(event: 'result', listener: (result: EmitResult) => any): this;
+  /**
+   * Adds a listener for the 'statusChanged' event that is called with a Diagnostic whenever the
+   * status of the current watch process changes.
+   *
+   * @param event    the 'statusChanged' event
+   * @param listener the listener to be called with the Diagnostic.
+   */
+  public on(event: 'statusChanged', listener: (message: Diagnostic) => any): this;
+  public on(event: string, listener: (...args: any[]) => any): this {
+    this.eventEmitter.on(event, listener);
+    return this;
+  }
+
+  /**
+   * Adds a once-only listener for the 'result' event that is called with EmitResult.
+   *
+   * @param event    the 'result' event
+   * @param listener the listener to be called with the EmitResult.
+   */
+  public once(event: 'result', listener: (result: EmitResult) => any): this;
+  /**
+   * Adds a once-only listener for the 'statusChanged' event that is called with a Diagnostic
+   * whenever the status of the current watch process changes.
+   *
+   * @param event    the 'statusChanged' event
```

Comment:
not sure I understand why was this needed as part of this change? Also feels like it might be risky...

Reply:
This is 100% motivated by the desire/need to be able to unit-test `watch` mode to ensure it actually does watch.